tests/unit/test_parsers.py

```diff
@@ -1081,6 +1081,100 @@ def test_response_metadata_errors_alternate_form_json_protocol(self):
         self.assertEqual(parsed['Error']['Message'], 'this is a message')
         self.assertEqual(parsed['Error']['Code'], 'ValidationException')
 
+    def test_response_with_query_error_for_json_protocol(self):
+        parser = parsers.JSONParser()
+        response = {
+            "body": b"""
+                {"__type":"amazon.foo.validate#ValidationException",
+                 "message":"this is a message"}
+                """,
+            "status_code": 400,
+            "headers": {
+                "x-amzn-requestid": "request-id",
+                "x-amzn-query-error": "AWS.SimpleQueueService.NonExistentQueue;Sender",
+            },
+        }
+        parsed = parser.parse(response, None)
+        # Even (especially) on an error condition, the
+        # ResponseMetadata should be populated.
+        self.assertIn('ResponseMetadata', parsed)
+        self.assertEqual(parsed['ResponseMetadata']['RequestId'], 'request-id')
+
+        self.assertIn('Error', parsed)
+        self.assertEqual(parsed['Error']['Message'], 'this is a message')
+        self.assertEqual(
+            parsed['Error']['Code'], 'AWS.SimpleQueueService.NonExistentQueue'
+        )
+
+    def test_response_with_invalid_query_error_for_json_protocol(self):
+        parser = parsers.JSONParser()
+        response = {
+            "body": b"""
+                {"__type":"amazon.foo.validate#ValidationException",
+                 "message":"this is a message"}
+                """,
+            "status_code": 400,
+            "headers": {
+                "x-amzn-requestid": "request-id",
+                "x-amzn-query-error": "AWS.SimpleQueueService.NonExistentQueue;sender;400",
+            },
+        }
+        parsed = parser.parse(response, None)
+        # Even (especially) on an error condition, the
+        # ResponseMetadata should be populated.
```

Comment:
I'm not sure this comment is needed in every test. Was there something novel we were trying to note about this behavior?

Reply:
Thanks for catching that. This comment is indeed not needed for every test case. I will remove them accordingly and only leave one in the first test case as suggested